### PR TITLE
Use only one component for images and use media API to resize them

### PR DESCRIPTION
**Pull request by VincentCATILLON**
## Detailed purpose of the PR

This PR uses media API to optimize usage (by resize) of images everywhere.

Btw, we only use 1 image component right now, instead of 4 different before: 
- `ImageGradient` (local)
- `ImageBackground` (react-native)
- `Image` (local)
- `Image` (react-native)

### Testing strategy

- [x] I ensured my code is tested: 
- [x] I took care to mention breaking changes and extra libs if appropriate: `Image` and `ImageGradient` components have been removed

### CODE REVIEW

**Comment by VincentCATILLON** [comment]
Idea: use a component instead of a helper

File: src/modules/image.js [hunk 1]
```diff
@@ -31,3 +35,22 @@ export const getImageDimensions = (
...
+export const getResizedImage = (
+  url: string,
+  {maxWidth, maxHeight, resizeMode}: {maxWidth?: number, maxHeight?: number, resizeMode?: 'contain'}
+) => {
+  if (!maxWidth && !maxHeight) {
+    return url;
+  }
+
+  const queryParams = buildUrlQueryParams({
+    ...(maxWidth ? {w: maxWidth} : {}),
```
Comment (Fausto95) [inline src/modules/image.js]: these spreads could be turned into something more readable

Comment (VincentCATILLON) [reply src/modules/image.js]: I agree with you, i'll make a component and move this logic into this one and make it easier to read btw

File: src/components/resources-browser.js [hunk 2]
```diff
@@ -97,37 +100,18 @@ class ResourcesBrowser extends React.PureComponent<Props> {
...
-                  )}
+                  <Preview
+                    type={resource.type}
+                    source={{uri: resource.poster}}
+                    hasOverlay={!isSelected}
+                    iconWidth={20}
+                    iconHeight={20}
+                    isIconVisible={!isSelected}
+                    style={styles.thumbnail}
+                  />
```
Comment (VincentCATILLON) [inline src/components/resources-browser.js]: This thing has been refactored due to duplicated code with `Preview` component

File: src/modules/image.js [hunk 3]
```diff
@@ -1,33 +1,44 @@
 // @flow strict
 
-export type Dimensions = {|
-  width: number,
-  height: number
-|};
-
-export const getImageDimensions = (
```
Comment (VincentCATILLON) [inline src/modules/image.js]: This function has been removed because it's unused now

File: src/modules/image.js (hunk 1 above)
Comment (VincentCATILLON) [reply src/modules/image.js]: done @Fausto95 :) https://github.com/CoorpAcademy/mobile/pull/391/files?file-filters%5B%5D=.js#diff-62408838b1f3a42b337cd185b30fd9b3R21

File: src/components/button-sticky.stories.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ import {storiesOf} from '@storybook/react-native';
 import {handleFakePress} from '../utils/tests';
 import {Component as ButtonSticky} from './button-sticky';
 
-storiesOf('Button Footer', module)
+storiesOf('ButtonSticky', module)
```
Comment (VincentCATILLON) [inline src/components/button-sticky.stories.js]: Story name not up to date

File: src/components/catalog.stories.js [hunk 5]
```diff
@@ -19,7 +19,11 @@ if (__TEST__) {
 const sections = createSections();
 const sectionsWithCardsRef = sections.map((section, index) => ({
   ...section,
-  cardsRef: (index === 0 && ['foo']) || (index === 1 && ['bar']) || (index === 2 && []) || undefined
+  cardsRef:
+    (index === 0 && ['foo', 'bar']) ||
+    (index === 1 && ['bar', 'foo']) ||
+    (index === 2 && []) ||
+    undefined
```
Comment (VincentCATILLON) [inline src/components/catalog.stories.js]: just to see multiple cards next to each other

File: src/components/brand-theme-provider.stories.js [hunk 6]
```diff
@@ -17,7 +17,11 @@ storiesOf('BrandThemeProvider', module).add('Default', () => (
             <Text style={{color: brandTheme.colors.primary}}>Primary color</Text>
             <Text>Content category name: {brandTheme.contentCategoryName}</Text>
             <Text>Name: {brandTheme.name}</Text>
-            <Image source={{uri: brandTheme.images['logo-mobile']}} style={{width: 200}} />
+            <Image
```
Comment (chrisdugne) [inline src/components/brand-theme-provider.stories.js]: ImageBackground ?

Comment (VincentCATILLON) [reply src/components/brand-theme-provider.stories.js]: I can use it yes, but it's only to see brand theme context, so it's not really important

Comment (VincentCATILLON) [reply src/components/brand-theme-provider.stories.js]: IMO, we should use a text component to display the value instead of the image

**Pull request merged**